Assets/Global/Scripts/Player/Player.cs

```diff
@@ -1,42 +1,42 @@
 using System;
 using UnityEngine;
+using UnityEngine.Serialization;
 
 public class Player : MonoBehaviour
 {
-    
-    public float jumpforce = 2f;
+    [Header("Settings")]
+    public float jumpForce = 2f;
     public float speed = 5f;
+    public float airBornMovementFactor = 0.5f;
     public int doubleJumps = 1;
     public float glideDrag = 2f;
     
-    [HideInInspector]
-    public BaseState currentState;
-    [HideInInspector] 
-    public int currentJumps = 0;
-    [HideInInspector]
-    public float horizontalInput;
-    [HideInInspector]
-    public float verticalInput;
-    public Rigidbody playerRb;
-    [HideInInspector]
-    public bool isGrounded;
+    [Header("References")]
+    [FormerlySerializedAs("playerRb")] 
+    public Rigidbody rb;
     public Transform orientation;
-
+    
+    [HideInInspector] public int currentJumps = 0;
+    [HideInInspector] public float horizontalInput;
+    [HideInInspector] public float verticalInput;
+    [HideInInspector] public bool isGrounded;
+    [HideInInspector] public PlayerStates states;
+    private StateBase currentState;
+    
     [Header("Debugging")]
-    public string currentStateName;
+    [SerializeField] private string currentStateName = "none";
 
     void Start()
     {
-        // playerRb = GetComponent<Rigidbody>();
-        SetState(new IdleState(this));
+        states = new PlayerStates(this);
```

Comment:
added a different way of settings states. if you scroll you will find it. 
It basically makes it so each state is not constantly created over and over again.
Instead it now reuses the same one each time.

The second benefit of doing it this way instead of a dictionairy is that this is not prone to spelling mistakes. 
This also allows us the rename states which is really hard to do if its string based.